ecr-login/cache/file.go

```diff
@@ -98,6 +98,9 @@ func (f *fileCredentialCache) fullFilePath() string {
 // file access. There is not guarantee here for handling multiple writes at once since there is no out of process locking.
 func (f *fileCredentialCache) save(registryCache *RegistryCache) error {
 	defer log.Flush()
+	if _, err := os.Stat(f.path); err != nil {
```

Comment:
> Are we okay with having the log and the cache file stored in different directories?

As we spoke about this earlier, I think it'd be better to abstract out the logs anyway so there's not a concern about this.  It'd be good to take a look at how the AWS SDK for Go does logging; it allows you to pass in a logger when you want to use it and configure it however you want.

> If customers are using both the credential helper and the cli, without it make sense for us to store and use the cache file from the same directory?

This is probably the most important point.  I don't think it's harmful to reuse the same cache file, but I'm also not sure it's necessary; either way, caching will occur and will lower the rate of calls to `GetAuthorizationToken`.